fieldservice/models/fsm_equipment.py

```diff
@@ -71,6 +71,13 @@ def advance_stage(self):
         self.stage_id = next_stage
         self._onchange_stage_id()
 
+    def previous_stage(self):
+        seq = self.stage_id.sequence
+        next_stage = self.env['fsm.stage'].search(
+            [('stage_type', '=', 'location'), ('sequence', '=', seq - 1)])
```

Comment:
@osi-scampbell It might return nothing, especially when the sequence is 12, 20, 30, 40...

You want to search with sequence < seq, ordered by sequence descending and with limit = 1 to get the previous.